src/workflows/airqo_etl_utils/ml_utils.py

```
@@ -694,82 +781,360 @@ def flag_rule_based_faults(df: pd.DataFrame) -> pd.DataFrame:
+        """
+        try:
+            bucket = FaultDetectionUtils._get_model_bucket()
+        except ValueError as exc:
+            logger.warning(str(exc))
+            return None
+
+        model_path = FaultDetectionUtils._get_isolation_forest_model_path()
+        try:
+            model = FaultDetectionUtils._load_model_object(
+                source_file=model_path,
+                local_cache_path=f"/tmp/{model_path}",
```

Ruff (0.15.11)

[error] Probable insecure usage of temporary file or directory: "/tmp/"

(S108)